Benchmark is out of date

When running `npm run benchmark` in benchmarks folder, I receive the following error:

```
> node autocannon-layered-loader.js

node:internal/errors:496
    ErrorCaptureStackTrace(err);
    ^

Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/Users/yagiz/Developer/layered-loader/dist/lib/LoadingOperation.js' imported from /Users/yagiz/Developer/layered-loader/benchmark/layered-loader/caches.js
    at new NodeError (node:internal/errors:405:5)
    at finalizeResolution (node:internal/modules/esm/resolve:224:11)
    at moduleResolve (node:internal/modules/esm/resolve:837:10)
    at defaultResolve (node:internal/modules/esm/resolve:1035:11)
    at DefaultModuleLoader.resolve (node:internal/modules/esm/loader:269:12)
    at DefaultModuleLoader.getModuleJob (node:internal/modules/esm/loader:153:32)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:76:33)
    at link (node:internal/modules/esm/module_job:75:36) {
  code: 'ERR_MODULE_NOT_FOUND'
}

```

--- a/benchmark/autocannon-layered-loader.js
+++ b/benchmark/autocannon-layered-loader.js
@@ -1,9 +1,9 @@
-import autocannon from 'autocannon'
+const autocannon = require('autocannon')
 
-import { createFastifyApp, startApp } from './common/setup.js'
-import { createLoadingOperation } from './layered-loader/caches.js'
-import { getSpecs } from './common/results.js'
-import { autocannonConfig } from './common/autocannonConfig.js'
+const { createFastifyApp, startApp } = require('./common/setup.js')
+const { createLoadingOperation } = require('./layered-loader/caches.js')
+const { getSpecs } = require('./common/results.js')
+const { autocannonConfig } = require('./common/autocannonConfig.js')
 
 const app = createFastifyApp()
 const { cache, redis } = createLoadingOperation()

--- a/benchmark/common/autocannonConfig.js
+++ b/benchmark/common/autocannonConfig.js
@@ -1,4 +1,4 @@
-export const autocannonConfig = {
+const autocannonConfig = {
   url: 'http://localhost:3000',
   workers: 8,
   connections: 4000,
@@ -8,4 +8,8 @@ export const autocannonConfig = {
     connections: 10,
     duration: 5,
   },
+}
+
+module.exports = {
+  autocannonConfig,
 }

--- a/benchmark/common/db/dbConfig.js
+++ b/benchmark/common/db/dbConfig.js
@@ -1,4 +1,4 @@
-export const dbConfig = {
+const dbConfig = {
   client: 'postgres',
   connection: {
     adapter: 'postgresql',
@@ -8,4 +8,8 @@ export const dbConfig = {
     port: 25432,
     database: 'cache_test',
   },
+}
+
+module.exports = {
+  dbConfig,
 }

--- a/benchmark/common/db/dbMigrate.js
+++ b/benchmark/common/db/dbMigrate.js
@@ -1,5 +1,5 @@
-import knex from 'knex'
-import { dbConfig } from './dbConfig.js'
+const knex = require('knex')
+const { dbConfig } = require('./dbConfig.js')
 
 const knexInstance = knex({
   ...dbConfig,

--- a/benchmark/common/db/dbSeed.js
+++ b/benchmark/common/db/dbSeed.js
@@ -1,8 +1,8 @@
-import knex from 'knex'
-import { dbConfig } from './dbConfig.js'
-import { UserRepository } from './repository.js'
+const knex = require('knex')
+const { dbConfig } = require('./dbConfig.js')
+const { UserRepository } = require('./repository.js')
 
-export async function seedWithData() {
+async function seedWithData() {
   const knexInstance = knex({
     ...dbConfig,
     migrations: {
@@ -25,4 +25,4 @@ export async function seedWithData() {
   console.log('Finished seeding')
 }
 
-await seedWithData()
+seedWithData()

--- a/benchmark/common/db/repository.js
+++ b/benchmark/common/db/repository.js
@@ -1,10 +1,14 @@
-import { AbstractRepository } from 'knex-repositories'
+const { AbstractRepository } = require('knex-repositories')
 
-export class UserRepository extends AbstractRepository {
+class UserRepository extends AbstractRepository {
   constructor(knex) {
     super(knex, {
       tableName: 'users',
       idColumn: 'id',
     })
   }
+}
+
+module.exports = {
+  UserRepository,
 }

--- a/benchmark/common/results.js
+++ b/benchmark/common/results.js
@@ -1,6 +1,6 @@
-import systemInformation from 'systeminformation'
+const systemInformation = require('systeminformation')
 
-export async function getSpecs() {
+async function getSpecs() {
   const cpuInfo = await systemInformation.cpu()
 
   return {
@@ -9,4 +9,8 @@ export async function getSpecs() {
       speed: `${cpuInfo.speed} GHz`,
     },
   }
+}
+
+module.exports = {
+  getSpecs,
 }

--- a/benchmark/common/setup.js
+++ b/benchmark/common/setup.js
@@ -1,17 +1,17 @@
-import { Redis } from 'ioredis'
-import fastify from 'fastify'
+const { Redis } = require('ioredis')
+const fastify = require('fastify')
 
-export const redisOptions = {
+const redisOptions = {
   host: 'localhost',
   port: 6379,
   password: 'sOmE_sEcUrE_pAsS',
 }
 
-export function createRedisConnection() {
+function createRedisConnection() {
   return new Redis(redisOptions)
 }
 
-export function createFastifyApp() {
+function createFastifyApp() {
   const app = fastify({
     logger: false,
     disableRequestLogging: true,
@@ -20,9 +20,16 @@ export function createFastifyApp() {
   return app
 }
 
-export function startApp(app) {
+function startApp(app) {
   app.listen({
     host: 'localhost',
     port: 3000,
   })
+}
+
+module.exports = {
+  createFastifyApp,
+  createRedisConnection,
+  startApp,
+  redisOptions,
 }

--- a/benchmark/layered-loader/caches.js
+++ b/benchmark/layered-loader/caches.js
@@ -1,9 +1,8 @@
-import { LoadingOperation } from '../../dist/lib/LoadingOperation.js'
-import { InMemoryCache } from '../../dist/lib/memory/index.js'
-import { RedisCache } from '../../dist/lib/redis/index.js'
-import { createRedisConnection } from '../common/setup.js'
+const { Loader } = require('../../dist/lib/Loader')
+const { RedisCache } = require('../../dist/lib/redis/')
+const { createRedisConnection } = require('../common/setup')
 
-export class DummyLoader {
+class DummyLoader {
   value
   name = 'Dummy loader'
   isCache = false
@@ -17,18 +16,23 @@ export class DummyLoader {
   }
 }
 
-export function createLoadingOperation() {
+function createLoadingOperation() {
   const redis = createRedisConnection()
   return {
     redis,
-    cache: new LoadingOperation([
-      new InMemoryCache({
+    cache: new Loader({
+      inMemoryCache: {
         ttlInMsecs: 5000,
-      }),
-      new RedisCache(redis, {
+      },
+      asyncCache: new RedisCache(redis, {
         ttlInMsecs: 60000,
       }),
-      new DummyLoader('value'),
-    ]),
+      dataSources: [new DummyLoader('value')],
+    }),
   }
+}
+
+module.exports = {
+  DummyLoader,
+  createLoadingOperation,
 }

--- a/benchmark/package.json
+++ b/benchmark/package.json
@@ -1,10 +1,11 @@
 {
   "name": "benchmark",
   "version": "1.0.0",
-  "type": "module",
+  "type": "commonjs",
   "scripts": {
     "build": "cd .. && npm run build",
-    "benchmark": "node autocannon-layered-loader.js",
+    "benchmark:full": "npm run build && npm run docker:start && npm run db:migrate && npm run db:seed && npm run benchmark:run && npm run docker:stop",
+    "benchmark:run": "node autocannon-layered-loader.js",
     "docker:start": "docker-compose -f docker-compose.yml up --build -d",
     "docker:stop": "docker-compose -f docker-compose.yml down",
     "db:migrate": "node common/db/dbMigrate.js",
@@ -26,5 +27,8 @@
   "devDependencies": {
     "@types/autocannon": "^7.9.0",
     "prettier": "^2.8.4"
+  },
+  "engines": {
+    "node": ">= 16 < 20"
   }
 }

--- a/package.json
+++ b/package.json
@@ -9,11 +9,7 @@
       "email": "kibertoad@gmail.com"
     }
   ],
-  "main": "dist/layered-loader",
-  "exports": {
-    "import": "./dist/layered-loader.js",
-    "require": "./dist/layered-loader.cjs"
-  },
+  "main": "dist/index.js",
   "types": "dist/index.d.ts",
   "scripts": {
     "build": "tsc",

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -1,7 +1,7 @@
 {
   "compilerOptions": {
     "outDir": "dist",
-    "module": "ES2022",
+    "module": "CommonJS",
     "target": "ES2022",
     "sourceMap": true,
     "declaration": true,

